ros_buildfarm/templates/devel/devel_job.xml.em

```diff
@@ -245,6 +256,106 @@ if pull_request:
     'publisher_xunit',
     pattern='catkin_workspace/test_results/**/*.xml',
 ))@
+@[if not pull_request]@
+@(SNIPPET(
+    'publisher_groovy-postbuild',
+    script='\n'.join([
+        '// COLLATE BUILD TEST RESULTS AND EXPORT BUILD HISTORY FOR WIKI',
+        'import jenkins.model.Jenkins',
+        'import hudson.FilePath',
+        '',
+        '@Grab(\'org.yaml:snakeyaml:1.17\')',
+        'import org.yaml.snakeyaml.Yaml',
+        'import org.yaml.snakeyaml.DumperOptions',
+        '',
+        'manager.listener.logger.println("# BEGIN SECTION: Collate test results for wiki.")',
+        '',
+        '// nr of builds to include in history',
+        'final num_build_hist = 5',
+        '',
+        'try {',
+        '  def data = [',
+        '    "history" : []',
+        '  ]',
+        '',
+        '  // gather info on tests of current build',
+        '  def tresult = manager.build.getAction(hudson.tasks.junit.TestResultAction.class)?.result',
+        '  if (tresult)',
+        '  {',
+        '    data.latest_build = [',
+        '      "skipped" : tresult.skipCount,',
+        '      "failed" : tresult.failCount,',
+        '      "total" : tresult.totalCount',
+        '    ]',
+        '  }',
+        '  else',
+        '  {',
+        '    manager.listener.logger.println("No test result action for last build, skipping gathering statistics for it.")',
+        '  }',
+        '',
+        '',
+        '  // get acces to the job of the running build',
+        '  def job_name = manager.build.getEnvironment(manager.listener).get(\'JOB_NAME\')',
+        '  manager.listener.logger.println("Collating test statistics for \'${job_name}\'.")',
+        '  def job = Jenkins.instance.getItem(job_name)',
+        '  if (job == null)',
+        '  {',
+        '    manager.listener.logger.println("No such job: \'${job_name}\'.")',
+        '    return',
```

Comment:
Interesting, just out of curiosity: did you ever run into this being the case during testing?

Reply:
No, I haven't. But I'm a rather defensive programmer most of the time. I can take it out if preferred.

Reply:
No need, better be safe than sorry :+1:

Reply:
Seems this does actually happen, see [Mdev__orocos_kinematics_dynamics__ubuntu_bionic_amd64](http://build.ros.org/job/Mdev__orocos_kinematics_dynamics__ubuntu_bionic_amd64/4/console) fi:

> No test result action for last build, skipping gathering statistics for it.

This seems to be caused by the `xUnit` plugin not finding any test reports:

```
[INFO] - [GoogleTest-1.6] - No test report file(s) were found with the pattern 'catkin_workspace/test_results/**/*.xml' relative to '/home/jenkins-agent/workspace/Mdev__orocos_kinematics_dynamics__ubuntu_bionic_amd64' for the testing framework 'GoogleTest-1.6'.  Did you enter a pattern relative to the correct directory?  Did you generate the result report(s) for 'GoogleTest-1.6'?
[WARNING] - No test reports found for the metric 'GoogleTest' with the resolved pattern 'catkin_workspace/test_results/**/*.xml'
[INFO] - Skipping the metric tool processing.
[INFO] - There are errors when processing test results.
[INFO] - Skipping tests recording.
```